ambry-store/src/main/java/com.github.ambry.store/BlobStore.java

```diff
@@ -683,6 +684,13 @@ public boolean isBootstrapInProgress() {
     return (new File(dataDir, BOOTSTRAP_FILE_NAME)).exists();
   }
 
+  @Override
+  public boolean isDecommissionInProgress() {
+    // note that, the decommission file will be removed by calling deleteStoreFiles() when replica is being dropped. We
+    // don't need to explicitly delete it. The file is also used for failure recovery to resume decommission process.
+    return (new File(dataDir, DECOMMISSION_FILE_NAME)).exists();
```

Comment:
decommission is per partition, does this work with more than on partition? if not, shouldn't we pass the partition name as part of the filename?

Reply:
Each store represents one partition. We put file into each store's directory (`dataDir`). This method is within BlobStore, when it is called, it indicates if the partition associated with this store is in decommission process. (I need to apologize for confusing naming for state transition. `Decommission` has two phases: `deactivation` and `disconnection`.  Any partition that is being decommissioned first gets deactivated and then gets disconnected.)